qiskit_ibm_runtime/runtime_job_v2.py

```diff
@@ -181,10 +181,13 @@ def _status_from_job_response(self, response: Dict) -> JobStatus:
         Returns:
             Job status.
         """
-        mapped_job_status = API_TO_JOB_STATUS[response["state"]["status"].upper()]
-        if mapped_job_status == "CANCELLED" and self._reason == "RAN TOO LONG":
-            mapped_job_status = "ERROR"
-        return mapped_job_status
+        api_status = response["state"]["status"].upper()
+        if api_status in API_TO_JOB_STATUS:
+            mapped_job_status = API_TO_JOB_STATUS[api_status]
+            if mapped_job_status == "CANCELLED" and self._reason == "RAN TOO LONG":
+                mapped_job_status = "ERROR"
```

Comment:
Maybe we could define constants

Reply:
Yeah we have an old issue - https://github.com/Qiskit/qiskit-ibm-runtime/issues/989 to avoid hard coding these job statuses. I'll follow up separately, I believe the server side is not returning an error code